pkg/domainutil/domain.go

```diff
@@ -0,0 +1,60 @@
+package domainutil
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	DefaultDomain = "astronomer.io"
+	LocalDomain   = "localhost"
+)
+
+var PRPreviewDomainRegex = regexp.MustCompile(`^(pr\d{4,6}).astronomer-dev\.io$`)
+
+func FormatDomain(domain string) string {
+	if strings.Contains(domain, "cloud") {
+		domain = strings.Replace(domain, "cloud.", "", 1)
+	} else if domain == "" {
+		domain = DefaultDomain
+	}
+
+	return domain
+}
+
+func isPrPreviewDomain(domain string) bool {
+	return PRPreviewDomainRegex.MatchString(domain)
+}
+
+func getPRSubDomain(domain string) (prSubDomain, restOfDomain string) {
+	if isPrPreviewDomain(domain) {
```

Comment:
nit: this is a duplicate check, since the caller already checks if it's a pr preview domain and then only calls this method